homeassistant/components/smtp/notify.py

```diff
@@ -267,7 +268,7 @@ def _build_html_msg(text, html, images):
         name = os.path.basename(atch_name)
         try:
             with open(atch_name, "rb") as attachment_file:
-                attachment = MIMEImage(attachment_file.read(), filename=name)
+                attachment = _build_attachment(attachment_file.read(), atch_name)
```

Comment:
Hi @esev, looks good to me, but, as I see the code now being equal for html and multipart msgs, if the logic is the same and the error handling for FileNotFoundError is also the same, why not refactoring all in an **`_attach_file(msg, atch_name, content_id)`**? (instead of the `_build_attachment(...)`)

Reply:
Yes, good call. That's a cleaner solution. Thanks.